src/particle.cc

```diff
@@ -1,13 +1,13 @@
 #include "particle.h"
 #include "factory.h"
-#include "particle_base.h"
+#include "particle.h"
 
 // Particle2D (2 Dim)
-static Register<mpm::ParticleBase<2>, mpm::Particle<2>, mpm::Index,
+static Register<mpm::Particle<2>, mpm::Particle<2>, mpm::Index,
                 const Eigen::Matrix<double, 2, 1>&>
     particle2d("P2D");
 
 // Particle3D (3 Dim)
-static Register<mpm::ParticleBase<3>, mpm::Particle<3>, mpm::Index,
+static Register<mpm::Particle<3>, mpm::Particle<3>, mpm::Index,
```

Comment:
Can you explain how we can register fluid particle and two-phase particles with this constructor?

Reply:
There is no difference in the constructor. We plan to store the particle type as a member variable

Reply:
Then, are we planning to read it from .json and assign the particle type when calling initialise_particles()?

Reply:
Yes, like we do now, we initialize the type appropriately from JSON object.